doc/languages-frameworks/vim.section.md

```diff
@@ -154,6 +154,43 @@ in
 }
 ```
 
+### Specificities for some plugins
+#### Tree sitter
+
+By default `nvim-treesitter` encourages you to download, compile and install
+the required tree-sitter grammars at run time with `:TSInstall`. This works
+poorly on NixOS.  Instead, to install the `nvim-treesitter` plugins with a set
+of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
+
+```nix
+{ pkgs, ... }:
+{
+  environment.systemPackages = [
+    (
+      pkgs.neovim.override {
+        configure = {
+          packages.myPlugins = with pkgs.vimPlugins; {
+            start = [
+              (
+                nvim-treesitter.withPlugins (
+                  plugins: with plugins; [
+                    tree-sitter-nix
+                    tree-sitter-python
+                  ]
+                )
+              )
+            ];
+          };
+        };
+      }
+    )
+  ];
+}
```

Comment:
```suggestion
{ pkgs, ... }:
{
  environment.systemPackages = [
    (pkgs.neovim.override {
      configure = {
        packages.myPlugins = with pkgs.vimPlugins; {
          start = [
            (nvim-treesitter.withPlugins (
              plugins: with plugins; [
                tree-sitter-nix
                tree-sitter-python
              ]
            ))
          ];
        };
      };
    })
  ];
}
```

Reply:
I formatted that with nixpkgs-fmt so I refuse bike-shedding.

Reply:
nixpkgs-fmt does not format everything the same if you give it different inputs. I just condensed the lines so that the example is shorter.

Reply:
done